flink-container/src/test/java/org/apache/flink/container/entrypoint/StandaloneJobClusterEntryPointTest.java

```diff
@@ -59,6 +66,60 @@ public void testDontOverwriteExecutionMode() {
 		assertThat(getExecutionMode(configuration), equalTo(ExecutionMode.NORMAL));
 	}
 
+	@Test
+	public void configuredJobIDTakesPrecedenceWithHA() {
+		Optional<JobID> jobID = Optional.of(JobID.generate());
+
+		Configuration globalConfiguration = new Configuration();
+		globalConfiguration.setString(HighAvailabilityOptions.HA_MODE, "zookeeper");
+
+		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.createJobIdForCluster(
+				jobID,
+				globalConfiguration);
+
+		assertThat(jobIdForCluster, is(jobID.get()));
+	}
+
+	@Test
+	public void configuredJobIDTakesPrecedenceWithoutHA() {
+		Optional<JobID> jobID = Optional.of(JobID.generate());
+
+		Configuration globalConfiguration = new Configuration();
+
+		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.createJobIdForCluster(
+				jobID,
+				globalConfiguration);
+
+		assertThat(jobIdForCluster, is(jobID.get()));
+	}
+
+	@Test
+	public void jobIDdefaultsToZeroWithHA() {
+		Optional<JobID> jobID = Optional.empty();
+
+		Configuration globalConfiguration = new Configuration();
+		globalConfiguration.setString(HighAvailabilityOptions.HA_MODE, "zookeeper");
```

Comment:
one could encapsulate this call into `enableHighAvailability(Configuration)` and that way reduce duplication.

Reply:
Done